Foundation/NSNumberFormatter.swift

```diff
@@ -7,18 +7,34 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
+import CoreFoundation
 
 public class NSNumberFormatter : NSFormatter {
     
-    public override init() { NSUnimplemented() }
-    public required init?(coder aDecoder: NSCoder) {
-        NSUnimplemented()
+    typealias CFType = CFNumberFormatterRef
+    internal var __cfObject: CFType?
+    internal var _cfObject: CFType {
+        get {
+            if let obj = __cfObject {
+                return obj
+            } else {
+                #if os(OSX) || os(iOS)
+                    let numberStyle = CFNumberFormatterStyle(rawValue: CFIndex(self.numberStyle.rawValue))!
+                #else
+                    let numberStyle = CFNumberFormatterStyle(self.numberStyle.rawValue)
+                #endif
+                
+                let obj = CFNumberFormatterCreate(kCFAllocatorSystemDefault, locale._cfObject, numberStyle)
+                _setFormatterAttributes(obj)
+                __cfObject = obj
+                return obj
+            }
+        }
```

Comment:
Two things here: 

1. We use the `_cfObject` method for types which are bridged. I think that the CFNumberFormatter/NSNumberFormatter relationship could instead be 'has-a', as in NSNumberFormatter has a CFNumberFormatterRef. This means we don't have to worry about things like matching up ivar layout. So I would call this internal var something besides _cfObject to avoid confusion.

2. Whatever we call the internal var backing the internal var, it'd be nice if it was not the same name with just an additional underscore... I feel like it's too easy to mess that up and use it accidentally somewhere in the rest of the class.